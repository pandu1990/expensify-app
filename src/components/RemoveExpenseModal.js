import React from 'react';
import Modal from 'react-modal';

const RemoveExpenseModal = ({ showModal, closeModal, removeExpense }) => (
  <Modal ariaHideApp={false} isOpen={showModal} contentLabel="Remove Expense" onRequestClose={closeModal} closeTimeoutMS={200} className="modal">
    <h3 className="modal__title">Remove Expense</h3>
    <p className="modal__body">Are you sure you want to remove this expense?</p>
    <div className="modal__actions">
      <button
        className="button button--secondary button--modal"
        onClick={() => {
          removeExpense();
          closeModal();
        }}
      >
        Yes
      </button>
      <button className="button button--modal" onClick={closeModal}>
        No
      </button>
    </div>
  </Modal>
);

export default RemoveExpenseModal;
