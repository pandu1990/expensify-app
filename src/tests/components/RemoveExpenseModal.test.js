import React from 'react';
import { shallow } from 'enzyme';
import { test, expect, jest, beforeAll } from '@jest/globals';
import RemoveExpenseModal from '../../components/RemoveExpenseModal';

let wrapper, closeModalSpy, removeExpenseSpy;
beforeEach(() => {
  closeModalSpy = jest.fn();
  removeExpenseSpy = jest.fn();
  wrapper = shallow(<RemoveExpenseModal showModal={true} closeModal={closeModalSpy} removeExpense={removeExpenseSpy} />);
});

test('should render RemoveExpenseModal correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should close modal when No button is clicked', () => {
  wrapper.find('button').at(1).simulate('click');
  expect(closeModalSpy).toHaveBeenCalled();
  expect(removeExpenseSpy).not.toHaveBeenCalled();
});

test('should close modal and call remove expense when Yes button is clicked', () => {
  wrapper.find('button').at(0).simulate('click');
  expect(closeModalSpy).toHaveBeenCalled();
  expect(removeExpenseSpy).toHaveBeenCalled();
});
