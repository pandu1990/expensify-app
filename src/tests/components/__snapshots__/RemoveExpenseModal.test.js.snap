// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RemoveExpenseModal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentElement={[Function]}
  contentLabel="Remove Expense"
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
    Remove Expense
  </h3>
  <p
    className="modal__body"
  >
    Are you sure you want to remove this expense?
  </p>
  <div
    className="modal__actions"
  >
    <button
      className="button button--secondary button--modal"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="button button--modal"
      onClick={[MockFunction]}
    >
      No
    </button>
  </div>
</Modal>
`;
